guides/writing-addons/deprecations.md

```diff
@@ -0,0 +1,78 @@
+There will come a day when you need to make breaking changes to your addon. How can you guide your users through the changes? Follow these steps to create deprecation warnings, give your users enough information to resolve them, and test that your deprecations work as expected.
+
+## What is a deprecation warning?
+
+A deprecation warning is a message printed to the console that informs a developer that they need to refactor part of their app to accommodate upcoming breaking changes to a library they use.
+
+## When to write a deprecation warning
+
+Deprecation warnings should be written whenever you know that you want to remove or change a feature of your addon.
+
+You can create deprecations at any time, but to provide the best experience for other developers, it's a good idea to create a deprecation only once you have a workaround available.
+
+For example, a workaround could be instructions about how to use your other existing features to accomplish the same thing, or a new API method that developers should migrate to.
+
+## How to make a deprecation warning
+
+You can create a deprecation warning that looks and works the same as the deprecations used by Ember itself:
+
+```js
+import { deprecate } from '@ember/application/deprecations';
+// ...
+
+yourDeprecatedMethod() {
+    deprecate('You used yourDeprecatedMethod in my-addon-name, which is deprecated. Here are instructions to resolve this...', 
+        false,
+        {
+            id: 'your-deprecation-id',
+            until: '2.0.0',
+            url: 'link-to-your-deprecations-instructions'
+        }
+    );
+    // code that should run for the deprecated method
+}
+// ...
+```
+
+The most important reason to use Ember's built in [`deprecate`](https://api.emberjs.com/ember/release/functions/@ember%2Fapplication%2Fdeprecations/deprecate) method is that the warning will be silenced in production.
+If a customer opens up the console for an app that uses your addon, they shouldn't see anything!
+Developers can configure whether it shows up in other build environments as well, like testing and local development.
+
+## Testing the deprecation warning
+
+Whenever you add a deprecation, it's important to add a couple test cases that would trigger the warning. You should also write tests for the "new" way to use the addon.
```

Comment:
Not sure how to phrase it, but if you are testing deprecated behavior its really important to ensure that your addon's own tests do not trigger the deprecation outside of `assert.expectDeprection` blocks. Otherwise the addons own test console gets very very hard to use.